pandas/core/indexers.py

```diff
@@ -0,0 +1,211 @@
+"""
+Low-dependency indexing utilities.
+"""
+import numpy as np
+
+from pandas.core.dtypes.common import is_list_like
+from pandas.core.dtypes.generic import ABCIndexClass, ABCSeries
+
+
+# -----------------------------------------------------------
+# Indexer Identification
+
+def is_list_like_indexer(key) -> bool:
+    # allow a list_like, but exclude NamedTuples which can be indexers
+    return is_list_like(key) and not (isinstance(key, tuple) and type(key) is not tuple)
+
+
+def is_scalar_indexer(indexer, arr_value) -> bool:
```

Comment:
if possible to type pls do so (may be tricky)

I would type with Any explicity if it’s truly that
so we know this is done (meaning it has been typed)

possible to use reveal_type to figure things out here

Reply:
ATM I've added types for everything I had figured out with sufficient certainty.  I'd rather leave it clearly-unfinished so it gets more attention than put Any if we can be more specific

Reply:
can you fix this doc-sring in followup and type as much as possible here